cpp/src/arrow/util/neon_util.h

```diff
@@ -21,6 +21,8 @@ namespace arrow {
 
 #if defined(__aarch64__) || defined(__AARCH64__)
 
+#define ARROW_HAVE_NEON 1
```

Comment:
Just for the record, is Neon always available with Aarch64?

Reply:
Neon is mandatory for Armv8-A. Arm server CPU is based on Armv8-A.

Strictly speaking, there may be some special Cortex-A AArch64 implementations don't support Neon(see quote below). They are not targeting server market.

>https://developer.arm.com/docs/den0024/a/aarch64-floating-point-and-neon
_Both floating-point and NEON are required in all standard ARMv8 implementations. However, implementations targeting specialized markets may support the following combinations:
No NEON or floating-point.
Full floating-point and SIMD support with exception trapping.
Full floating-point and SIMD support without exception trapping._

Reply:
Ok, thank you.